routers/web/repo/release.go

```diff
@@ -156,6 +157,13 @@ func Releases(ctx *context.Context) {
 	ctx.Data["CanCreateBranch"] = false
 	ctx.Data["HideBranchesInDropdown"] = true
 
+	// repo/release_tag_header.tmpl will render sub menu depending on the different permission
```

Comment:
I still do not understand why the "different permission" matters.

Why not just render it without permission check?

And even worse, in the future, are you sure all "PrepareForRepoSubMenu" functions need the same permission check?

Reply:
If you open `repo/release_tag_header.tmpl` you will find the sub menu will not always display which depends on the permission check. That means if user have release permission, viewing tags will be redirected to release list page. The release list page have no sub menu. If users have no release permission but have code permission, viewing tags will just display a tag list. In this page, there is the sub menu.

If we don't do the permission check and always load sub menu, it may be a waste for the release list page.

Reply:
9ec4bdd1f1 removed all the unnecessary permission check on routers and templates.